qiime2/metadata/metadata.py

```diff
@@ -362,6 +386,43 @@ def __init__(self, dataframe):
         super().__init__(dataframe.index)
 
         self._dataframe, self._columns = self._normalize_dataframe(dataframe)
+        self.contains_renamed_columns = False
+
+    @property
+    def hash(self):
```

Comment:
Is `hash` the right word here? To me this seems more like a metadata `.id` property, where the id is either a UUID (in the case of Artifact-sourced MD), or a hash of the TSV representation of MD.